gtest-parallel

```diff
@@ -424,4 +470,6 @@ if options.print_test_times:
               if times.get_test_time(test_binary, test) is not None)
   for (time_ms, test_binary, test) in ts:
     print "%8s %s" % ("%dms" % time_ms, test)
+if options.dump_json_test_results:
+  test_results.dump_to_file(options.dump_json_test_results)
```

Comment:
Can you hold on to this file handle for the duration of the tests? CollectTestResults() could take dump_json_test_results as a parameter and .dump_to_file could be .write() or something (and the no-op test_results could have an empty .write()).